From f718361b95258244df1d26b5f15d3a8098259939 Mon Sep 17 00:00:00 2001
From: Pavel Vomacka <pvomacka@redhat.com>
Date: Fri, 26 Aug 2016 13:03:58 +0200
Subject: [PATCH] WebUI: Fix showing certificates issued by sub-CA

The cert-show command needs to be called with cacn option. Cacn option is
passed using URL attribute.

https://fedorahosted.org/freeipa/ticket/6238

Reviewed-By: Petr Vobornik <pvoborni@redhat.com>
---
 install/ui/src/freeipa/certificate.js | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/install/ui/src/freeipa/certificate.js b/install/ui/src/freeipa/certificate.js
index 232bdbf2fa95c3a68943539cd80129d481d8563a..e67c348b7edbd2e273f18ed1df40ef8b5b5e59c3 100755
--- a/install/ui/src/freeipa/certificate.js
+++ b/install/ui/src/freeipa/certificate.js
@@ -1543,6 +1543,7 @@ return {
             row_enabled_attribute: 'status',
             facet_groups: [exp.facet_group],
             facet_group: 'certificates',
+            additional_navigation_arguments: [ 'cacn' ],
             pagination: false,
             no_update: true,
             columns: [
@@ -1552,6 +1553,7 @@ return {
                     width: '90px'
                 },
                 'subject',
+                'cacn',
                 {
                     name: 'status',
                     width: '120px'
@@ -1645,6 +1647,7 @@ return {
                     fields: [
                         'serial_number',
                         'serial_number_hex',
+                        'cacn',
                         'subject',
                         {
                             name: 'issuer',
@@ -1772,6 +1775,10 @@ IPA.cert.details_facet = function(spec, no_init) {
         var command = that.details_facet_create_refresh_command();
         delete command.options.all;
         delete command.options.rights;
+
+        command.options = command.options || {};
+        $.extend(command.options, { cacn: that.state.cacn });
+
         return command;
     };
 
-- 
2.7.4

